metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -1199,6 +1199,126 @@ def create_tag(self, tag_name: str) -> str:
         # return urn
         return res["createTag"]
 
+    def _assertion_result_shared(self) -> str:
+        fragment: str = """
+             fragment assertionResult on AssertionResult {
+                 type
+                 rowCount
+                 missingCount
+                 unexpectedCount
+                 actualAggValue
+                 externalUrl
+                 nativeResults {
+                     value
+                 }
+                 error {
+                     type
+                     properties {
+                         value
+                     }
+                 }
+             }
+        """
+        return fragment
+
+
+    def _run_assertion_result_shared(self) -> str:
+        fragment: str = """
+            fragment runAssertionResult on RunAssertionResult {
+                assertion {
+                    urn
+                }
+                result {
+                    ... assertionResult
+                }
+            }
+        """
+        return fragment
+
+
+    def run_assertion(self, urn: str, save_result: bool = False):
+        graph_query: str = """
+            %s
+            mutation runAssertion($assertionUrn: String!, $saveResult: Boolean) {
+                runAssertion(urn: $assertionUrn, saveResult: $saveResult) {
+                    ... assertionResult
+                }
+            }
+        """ % (self._assertion_result_shared())
+
+        variables = {
+            "assertionUrn": urn,
+            "saveResult": save_result,
+        }
+
+        res = self.execute_graphql(
+            query=graph_query,
+            variables=variables,
+        )
+
+        # return urn
```

Comment:
Not the urn right? We are also returning the details of the run?

Reply:
Yep, just an unused comment. Remove